// enzyme-matchers/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _toBeChecked = require('./assertions/toBeChecked');

var _toBeChecked2 = _interopRequireDefault(_toBeChecked);

var _toBeDisabled = require('./assertions/toBeDisabled');

var _toBeDisabled2 = _interopRequireDefault(_toBeDisabled);

var _toBeEmptyRender = require('./assertions/toBeEmptyRender');

var _toBeEmptyRender2 = _interopRequireDefault(_toBeEmptyRender);

var _toContainMatchingElement = require('./assertions/toContainMatchingElement');

var _toContainMatchingElement2 = _interopRequireDefault(_toContainMatchingElement);

var _toContainMatchingElements = require('./assertions/toContainMatchingElements');

var _toContainMatchingElements2 = _interopRequireDefault(_toContainMatchingElements);

var _toContainExactlyOneMatchingElement = require('./assertions/toContainExactlyOneMatchingElement');

var _toContainExactlyOneMatchingElement2 = _interopRequireDefault(_toContainExactlyOneMatchingElement);

var _toContainReact = require('./assertions/toContainReact');

var _toContainReact2 = _interopRequireDefault(_toContainReact);

var _toExist = require('./assertions/toExist');

var _toExist2 = _interopRequireDefault(_toExist);

var _toHaveClassName = require('./assertions/toHaveClassName');

var _toHaveClassName2 = _interopRequireDefault(_toHaveClassName);

var _toHaveDisplayName = require('./assertions/toHaveDisplayName');

var _toHaveDisplayName2 = _interopRequireDefault(_toHaveDisplayName);

var _toHaveHTML = require('./assertions/toHaveHTML');

var _toHaveHTML2 = _interopRequireDefault(_toHaveHTML);

var _toHaveProp = require('./assertions/toHaveProp');

var _toHaveProp2 = _interopRequireDefault(_toHaveProp);

var _toHaveRef = require('./assertions/toHaveRef');

var _toHaveRef2 = _interopRequireDefault(_toHaveRef);

var _toHaveState = require('./assertions/toHaveState');

var _toHaveState2 = _interopRequireDefault(_toHaveState);

var _toHaveStyle = require('./assertions/toHaveStyle');

var _toHaveStyle2 = _interopRequireDefault(_toHaveStyle);

var _toHaveTagName = require('./assertions/toHaveTagName');

var _toHaveTagName2 = _interopRequireDefault(_toHaveTagName);

var _toHaveText = require('./assertions/toHaveText');

var _toHaveText2 = _interopRequireDefault(_toHaveText);

var _toIncludeText = require('./assertions/toIncludeText');

var _toIncludeText2 = _interopRequireDefault(_toIncludeText);

var _toHaveValue = require('./assertions/toHaveValue');

var _toHaveValue2 = _interopRequireDefault(_toHaveValue);

var _toMatchSelector = require('./assertions/toMatchSelector');

var _toMatchSelector2 = _interopRequireDefault(_toMatchSelector);

var _toMatchElement = require('./assertions/toMatchElement');

var _toMatchElement2 = _interopRequireDefault(_toMatchElement);

var _protectAssertion = require('./utils/protectAssertion');

var _protectAssertion2 = _interopRequireDefault(_protectAssertion);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree. *
 *
 * @providesModule enzyme-matchrs
 * 
 */

var assertions = {
  toBeChecked: _toBeChecked2.default,
  toBeDisabled: _toBeDisabled2.default,
  toBeEmptyRender: _toBeEmptyRender2.default,
  toContainMatchingElement: _toContainMatchingElement2.default,
  toContainMatchingElements: _toContainMatchingElements2.default,
  toContainExactlyOneMatchingElement: _toContainExactlyOneMatchingElement2.default,
  toContainReact: _toContainReact2.default,
  toExist: _toExist2.default,
  toHaveClassName: _toHaveClassName2.default,
  toHaveDisplayName: _toHaveDisplayName2.default,
  toHaveHTML: _toHaveHTML2.default,
  toHaveProp: _toHaveProp2.default,
  toHaveRef: _toHaveRef2.default,
  toHaveState: _toHaveState2.default,
  toHaveStyle: _toHaveStyle2.default,
  toHaveTagName: _toHaveTagName2.default,
  toHaveText: _toHaveText2.default,
  toIncludeText: _toIncludeText2.default,
  toHaveValue: _toHaveValue2.default,
  toMatchSelector: _toMatchSelector2.default,
  toMatchElement: _toMatchElement2.default
};

// Protect all assertions from being called with not-enzyme-wrappers.
Object.keys(assertions).forEach(function (key) {
  assertions[key] = (0, _protectAssertion2.default)(assertions[key]);
});

exports.default = assertions;
module.exports = exports['default'];
