// enzyme-matchers/lib/assertions/toContainMatchingElement.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _html = require('../utils/html');

var _html2 = _interopRequireDefault(_html);

var _name = require('../utils/name');

var _name2 = _interopRequireDefault(_name);

var _single = require('../utils/single');

var _single2 = _interopRequireDefault(_single);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * This source code is licensed under the MIT-style license found in the
 * LICENSE file in the root directory of this source tree. *
 *
 * @providesModule toContainMatchingElement
 * 
 */

function toContainMatchingElement(enzymeWrapper, selector) {
  var matches = enzymeWrapper.find(selector);
  var pass = matches.length > 0;
  var nodeName = (0, _name2.default)(enzymeWrapper);

  return {
    pass: pass,
    message: 'Expected <' + nodeName + '> to contain at least one element matching ' + (selector + ' but none were found.'),
    negatedMessage: 'Expected <' + nodeName + '> to not contain an element matching ' + (selector + ' but it did.'),
    contextualInformation: {
      actual: 'HTML Output of <' + nodeName + '>:\n ' + (0, _html2.default)(enzymeWrapper)
    }
  };
}

exports.default = (0, _single2.default)(toContainMatchingElement);
module.exports = exports['default'];
